src/themes/scripts/sw-controller.js

```diff
@@ -1,5 +1,13 @@
 window.addEventListener('load', function() {
   if (navigator.serviceWorker) {
+    if (window.BroadcastChannel) {
+      const updateChannel = new window.BroadcastChannel('precache-updates');
+      updateChannel.addEventListener('message', (event) => {
+        console.log(event);
```

Comment:
Given that this site uses real navigations (i.e. it's not a single-page app with the same page open indefinitely), I think we could get away with only prompting if the current page is updated. So something like:

```js
updateChannel.addEventListener('message', (event) => {
  if (event.data.payload.updatedUrl === location.href) {
    // Show toast...
  }
});
```

The wrinkle is that with the "clean" URLs option that's also in this PR, I'm not sure if there's an exact match of the current location and the cached URL, so you might need to append `.html` or something before you check.

That assumes that it's not worth bothering the user to refresh if a CSS file or some other subresource gets updated.